syntax = "proto3";

package orders;
option go_package = "/proto/orders;orders";

message Item {
  string product_id = 1;
  int32 quantity = 2;
}

message OrderRequest {
  string user_id = 1;
  repeated Item items = 2;
  string payment_method = 3;
}

message OrderResponse {
  string order_id = 1;
  string status = 2;
}

service OrderService {
  rpc CreateOrder(OrderRequest) returns (OrderResponse);
}
